opengrok-web/src/main/java/org/opengrok/web/PageConfig.java

```diff
@@ -446,28 +446,11 @@ public String canProcess() {
             if (isDir() && getResourcePath().length() > 1) {
                 files = getResourceFile().listFiles();
             }
+
             if (files == null) {
                 dirFileList = Collections.emptyList();
             } else {
-                List<String> listOfFiles;
-                if (env.getListDirsFirst()) {
-                    Arrays.sort(files, (f1, f2) -> {
```

Comment:
Wouldn't it make more sense to fix this comparator? E.g. something like this:
```java
if (f1.isDirectory() && !f2.isDirectory()) {
    return -1;
} else if (!f1.isDirectory() && f2.isDirectory()) {
    return 1;
} else {
    return f1.getName().compareTo(f2.getName());
}
```

I think file can be neither `isFile()` and `isDirectory()` as mentioned in the issue with the symbolic link which can cause inconsistency in the current implementation - even symmetry is not upheld.

Reply:
Good point ! I assume we want to sort the links with files.